docs: c-protobuf can't be compiled with g++5.3

In the root project directory,  executing `make` produces the following c-protobuf compile error:

```
...
honnef.co/go/unused                                16cf7f968ef7	[checkout 9057d575b3bb]
cmd github.com/client9/misspell/cmd/misspell                   	[built]
cmd github.com/cockroachdb/c-protobuf/cmd/protoc               	[error exit status 2]
github.com/cockroachdb/c-protobuf
# github.com/cockroachdb/c-protobuf
In file included from ../c-protobuf/datapiece.cc:38:0:
internal/src/google/protobuf/stubs/mathlimits.h: In static member function ‘static bool google::protobuf::MathLimits<double>::IsFinite(google::protobuf::MathLimits<double>::Type)’:
internal/src/google/protobuf/stubs/mathlimits.h:233:55: error: call of overloaded ‘isinf(const Type&)’ is ambiguous
   static bool IsFinite(const Type x) { return !isinf(x)  &&  !isnan(x); } \
                                                       ^
internal/src/google/protobuf/stubs/mathlimits.h:265:3: note: in expansion of macro ‘DECL_FP_LIMIT_FUNCS’
   DECL_FP_LIMIT_FUNCS \
   ^
internal/src/google/protobuf/stubs/mathlimits.h:269:1: note: in expansion of macro ‘DECL_FP_LIMITS’
 DECL_FP_LIMITS(double, DBL)
 ^
In file included from /usr/include/features.h:374:0,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/os_defines.h:39,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/c++config.h:482,
                 from /usr/include/c++/5/string:38,
                 from internal/src/google/protobuf/util/internal/datapiece.h:34,
                 from ../c-protobuf/datapiece.cc:31:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:201:1: note: candidate: int isinf(double)
 __MATHDECL_1 (int,isinf,, (_Mdouble_ __value)) __attribute__ ((__const__));
 ^
In file included from /usr/include/c++/5/random:38:0,
                 from internal/src/google/protobuf/map.h:47,
                 from internal/src/google/protobuf/struct.pb.h:29,
                 from ../c-protobuf/datapiece.cc:33:
/usr/include/c++/5/cmath:614:3: note: candidate: constexpr bool std::isinf(long double)
   isinf(long double __x)
   ^
/usr/include/c++/5/cmath:610:3: note: candidate: constexpr bool std::isinf(double)
   isinf(double __x)
   ^
/usr/include/c++/5/cmath:606:3: note: candidate: constexpr bool std::isinf(float)
   isinf(float __x)
   ^
In file included from ../c-protobuf/datapiece.cc:38:0:
internal/src/google/protobuf/stubs/mathlimits.h:233:70: error: call of overloaded ‘isnan(const Type&)’ is ambiguous
   static bool IsFinite(const Type x) { return !isinf(x)  &&  !isnan(x); } \
                                                                      ^
internal/src/google/protobuf/stubs/mathlimits.h:265:3: note: in expansion of macro ‘DECL_FP_LIMIT_FUNCS’
   DECL_FP_LIMIT_FUNCS \
   ^
internal/src/google/protobuf/stubs/mathlimits.h:269:1: note: in expansion of macro ‘DECL_FP_LIMITS’
 DECL_FP_LIMITS(double, DBL)
 ^
In file included from /usr/include/features.h:374:0,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/os_defines.h:39,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/c++config.h:482,
                 from /usr/include/c++/5/string:38,
                 from internal/src/google/protobuf/util/internal/datapiece.h:34,
                 from ../c-protobuf/datapiece.cc:31:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:234:1: note: candidate: int isnan(double)
 __MATHDECL_1 (int,isnan,, (_Mdouble_ __value)) __attribute__ ((__const__));
 ^
In file included from /usr/include/c++/5/random:38:0,
                 from internal/src/google/protobuf/map.h:47,
                 from internal/src/google/protobuf/struct.pb.h:29,
                 from ../c-protobuf/datapiece.cc:33:
/usr/include/c++/5/cmath:634:3: note: candidate: constexpr bool std::isnan(long double)
   isnan(long double __x)
   ^
/usr/include/c++/5/cmath:630:3: note: candidate: constexpr bool std::isnan(double)
   isnan(double __x)
   ^
/usr/include/c++/5/cmath:626:3: note: candidate: constexpr bool std::isnan(float)
   isnan(float __x)
   ^
In file included from ../c-protobuf/datapiece.cc:38:0:
internal/src/google/protobuf/stubs/mathlimits.h: In static member function ‘static bool google::protobuf::MathLimits<double>::IsNaN(google::protobuf::MathLimits<double>::Type)’:
internal/src/google/protobuf/stubs/mathlimits.h:234:51: error: call of overloaded ‘isnan(const Type&)’ is ambiguous
   static bool IsNaN(const Type x) { return isnan(x); } \
                                                   ^
internal/src/google/protobuf/stubs/mathlimits.h:265:3: note: in expansion of macro ‘DECL_FP_LIMIT_FUNCS’
   DECL_FP_LIMIT_FUNCS \
   ^
internal/src/google/protobuf/stubs/mathlimits.h:269:1: note: in expansion of macro ‘DECL_FP_LIMITS’
 DECL_FP_LIMITS(double, DBL)
 ^
In file included from /usr/include/features.h:374:0,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/os_defines.h:39,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/c++config.h:482,
                 from /usr/include/c++/5/string:38,
                 from internal/src/google/protobuf/util/internal/datapiece.h:34,
                 from ../c-protobuf/datapiece.cc:31:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:234:1: note: candidate: int isnan(double)
 __MATHDECL_1 (int,isnan,, (_Mdouble_ __value)) __attribute__ ((__const__));
 ^
In file included from /usr/include/c++/5/random:38:0,
                 from internal/src/google/protobuf/map.h:47,
                 from internal/src/google/protobuf/struct.pb.h:29,
                 from ../c-protobuf/datapiece.cc:33:
/usr/include/c++/5/cmath:634:3: note: candidate: constexpr bool std::isnan(long double)
   isnan(long double __x)
   ^
/usr/include/c++/5/cmath:630:3: note: candidate: constexpr bool std::isnan(double)
   isnan(double __x)
   ^
/usr/include/c++/5/cmath:626:3: note: candidate: constexpr bool std::isnan(float)
   isnan(float __x)
   ^
In file included from ../c-protobuf/datapiece.cc:38:0:
internal/src/google/protobuf/stubs/mathlimits.h: In static member function ‘static bool google::protobuf::MathLimits<double>::IsInf(google::protobuf::MathLimits<double>::Type)’:
internal/src/google/protobuf/stubs/mathlimits.h:235:51: error: call of overloaded ‘isinf(const Type&)’ is ambiguous
   static bool IsInf(const Type x) { return isinf(x); } \
                                                   ^
internal/src/google/protobuf/stubs/mathlimits.h:265:3: note: in expansion of macro ‘DECL_FP_LIMIT_FUNCS’
   DECL_FP_LIMIT_FUNCS \
   ^
internal/src/google/protobuf/stubs/mathlimits.h:269:1: note: in expansion of macro ‘DECL_FP_LIMITS’
 DECL_FP_LIMITS(double, DBL)
 ^
In file included from /usr/include/features.h:374:0,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/os_defines.h:39,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/c++config.h:482,
                 from /usr/include/c++/5/string:38,
                 from internal/src/google/protobuf/util/internal/datapiece.h:34,
                 from ../c-protobuf/datapiece.cc:31:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:201:1: note: candidate: int isinf(double)
 __MATHDECL_1 (int,isinf,, (_Mdouble_ __value)) __attribute__ ((__const__));
 ^
In file included from /usr/include/c++/5/random:38:0,
                 from internal/src/google/protobuf/map.h:47,
                 from internal/src/google/protobuf/struct.pb.h:29,
                 from ../c-protobuf/datapiece.cc:33:
/usr/include/c++/5/cmath:614:3: note: candidate: constexpr bool std::isinf(long double)
   isinf(long double __x)
   ^
/usr/include/c++/5/cmath:610:3: note: candidate: constexpr bool std::isinf(double)
   isinf(double __x)
   ^
/usr/include/c++/5/cmath:606:3: note: candidate: constexpr bool std::isinf(float)
   isinf(float __x)
   ^
In file included from ../c-protobuf/datapiece.cc:38:0:
internal/src/google/protobuf/stubs/mathlimits.h: In static member function ‘static bool google::protobuf::MathLimits<double>::IsPosInf(google::protobuf::MathLimits<double>::Type)’:
internal/src/google/protobuf/stubs/mathlimits.h:236:54: error: call of overloaded ‘isinf(const Type&)’ is ambiguous
   static bool IsPosInf(const Type x) { return isinf(x)  &&  x > 0; } \
                                                      ^
internal/src/google/protobuf/stubs/mathlimits.h:265:3: note: in expansion of macro ‘DECL_FP_LIMIT_FUNCS’
   DECL_FP_LIMIT_FUNCS \
   ^
internal/src/google/protobuf/stubs/mathlimits.h:269:1: note: in expansion of macro ‘DECL_FP_LIMITS’
 DECL_FP_LIMITS(double, DBL)
 ^
In file included from /usr/include/features.h:374:0,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/os_defines.h:39,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/c++config.h:482,
                 from /usr/include/c++/5/string:38,
                 from internal/src/google/protobuf/util/internal/datapiece.h:34,
                 from ../c-protobuf/datapiece.cc:31:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:201:1: note: candidate: int isinf(double)
 __MATHDECL_1 (int,isinf,, (_Mdouble_ __value)) __attribute__ ((__const__));
 ^
In file included from /usr/include/c++/5/random:38:0,
                 from internal/src/google/protobuf/map.h:47,
                 from internal/src/google/protobuf/struct.pb.h:29,
                 from ../c-protobuf/datapiece.cc:33:
/usr/include/c++/5/cmath:614:3: note: candidate: constexpr bool std::isinf(long double)
   isinf(long double __x)
   ^
/usr/include/c++/5/cmath:610:3: note: candidate: constexpr bool std::isinf(double)
   isinf(double __x)
   ^
/usr/include/c++/5/cmath:606:3: note: candidate: constexpr bool std::isinf(float)
   isinf(float __x)
   ^
In file included from ../c-protobuf/datapiece.cc:38:0:
internal/src/google/protobuf/stubs/mathlimits.h: In static member function ‘static bool google::protobuf::MathLimits<double>::IsNegInf(google::protobuf::MathLimits<double>::Type)’:
internal/src/google/protobuf/stubs/mathlimits.h:237:54: error: call of overloaded ‘isinf(const Type&)’ is ambiguous
   static bool IsNegInf(const Type x) { return isinf(x)  &&  x < 0; }
                                                      ^
internal/src/google/protobuf/stubs/mathlimits.h:265:3: note: in expansion of macro ‘DECL_FP_LIMIT_FUNCS’
   DECL_FP_LIMIT_FUNCS \
   ^
internal/src/google/protobuf/stubs/mathlimits.h:269:1: note: in expansion of macro ‘DECL_FP_LIMITS’
 DECL_FP_LIMITS(double, DBL)
 ^
In file included from /usr/include/features.h:374:0,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/os_defines.h:39,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/c++config.h:482,
                 from /usr/include/c++/5/string:38,
                 from internal/src/google/protobuf/util/internal/datapiece.h:34,
                 from ../c-protobuf/datapiece.cc:31:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:201:1: note: candidate: int isinf(double)
 __MATHDECL_1 (int,isinf,, (_Mdouble_ __value)) __attribute__ ((__const__));
 ^
In file included from /usr/include/c++/5/random:38:0,
                 from internal/src/google/protobuf/map.h:47,
                 from internal/src/google/protobuf/struct.pb.h:29,
                 from ../c-protobuf/datapiece.cc:33:
/usr/include/c++/5/cmath:614:3: note: candidate: constexpr bool std::isinf(long double)
   isinf(long double __x)
   ^
/usr/include/c++/5/cmath:610:3: note: candidate: constexpr bool std::isinf(double)
   isinf(double __x)
   ^
/usr/include/c++/5/cmath:606:3: note: candidate: constexpr bool std::isinf(float)
   isinf(float __x)
   ^
In file included from ../c-protobuf/datapiece.cc:39:0:
internal/src/google/protobuf/stubs/mathutil.h: In function ‘bool google::protobuf::internal::IsNan(T) [with T = double]’:
internal/src/google/protobuf/stubs/mathutil.h:60:21: error: call of overloaded ‘isnan(double&)’ is ambiguous
   return isnan(value);
                     ^
In file included from /usr/include/features.h:374:0,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/os_defines.h:39,
                 from /usr/include/x86_64-linux-gnu/c++/5/bits/c++config.h:482,
                 from /usr/include/c++/5/string:38,
                 from internal/src/google/protobuf/util/internal/datapiece.h:34,
                 from ../c-protobuf/datapiece.cc:31:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:234:1: note: candidate: int isnan(double)
 __MATHDECL_1 (int,isnan,, (_Mdouble_ __value)) __attribute__ ((__const__));
 ^
In file included from /usr/include/c++/5/random:38:0,
                 from internal/src/google/protobuf/map.h:47,
                 from internal/src/google/protobuf/struct.pb.h:29,
                 from ../c-protobuf/datapiece.cc:33:
/usr/include/c++/5/cmath:634:3: note: candidate: constexpr bool std::isnan(long double)
   isnan(long double __x)
   ^
/usr/include/c++/5/cmath:630:3: note: candidate: constexpr bool std::isnan(double)
   isnan(double __x)
   ^
/usr/include/c++/5/cmath:626:3: note: candidate: constexpr bool std::isnan(float)
   isnan(float __x)
   ^
make: *** [.bootstrap] Error 1
```

I am using g++ 5.3.0 on Ubuntu 14.04.  

[c-protobuf's circle.yml](https://github.com/cockroachdb/c-protobuf/blob/master/circle.yml) uses g++-6:

```
    - sudo apt-get install gcc-6 g++-6
```

But [CONTRIBUTING.md](https://github.com/cockroachdb/cockroach/blob/master/CONTRIBUTING.md) says:

> A C++ compiler that supports C++11 (GCC 4.9+ and clang 3.6+ are known to work).

After g++ is upgraded to 6.1.1, the c-protobuf compiler error is gone. So I think that the content related to g++ version in [CONTRIBUTING.md](https://github.com/cockroachdb/cockroach/blob/master/CONTRIBUTING.md)  should be updated.

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -3,16 +3,10 @@
 ### Getting and building
 
 1.  Install the following prerequisites, as necessary:
-	- A C++ compiler that supports C++11 (GCC 4.9+ and clang 3.6+ are
-	  known to work). On Mac OS X, Xcode should suffice.
-	- A [Go environment](http://golang.org/doc/code.html) with a
-	  64-bit version of Go 1.6.2. You can download the
-	  [Go binary](https://golang.org/dl/) directly from the official
-	  site. On OS X, you can also use [homebrew](http://brew.sh):
-	  `brew install go`. Be sure to set the `$GOPATH` and `$PATH`
-	  environment variables as described
-	  [here](https://golang.org/doc/code.html#GOPATH).
-	- Git 1.8+
+  - A C++ compiler that supports C++11. Note that GCC prior to 6.0 doesn't
+  work due to https://gcc.gnu.org/bugzilla/show_bug.cgi?id=48891
+  - A Go environment with a 64-bit version of Go 1.6.2.
+  - Git 1.8+.
 
 2.  Get the CockroachDB code:
 

